test/Interop/Cxx/value-witness-table/basic.swift

```diff
@@ -0,0 +1,122 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import Basic
+import StdlibUnittest
+
+var CXXDestructorTestSuite = TestSuite("CXXDestructor")
+
+protocol InitWithPtr {
+  init(value: UnsafeMutablePointer<Int32>!)
+}
+
+extension HasUserProvidedDestructor : InitWithPtr { }
+
+protocol InitWithBox {
+  init(box: HasUserProvidedDestructor)
+}
+
+extension EmptyDestructor : InitWithBox { }
+
+@inline(never)
+func createTypeWithUserProvidedDestructor(_ ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+}
+
+@inline(never)
+func createTypeWithEmptyDestructor(_ ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let empty = EmptyDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithNonTrivialImplicitDestructor(_ ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let implicit = HasNonTrivialImplicitDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithNonTrivialDefaultDestructor(_ ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let def = HasNonTrivialDefaultDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithGeneric<T : InitWithPtr>(_ ptr: UnsafeMutablePointer<Int32>,
+                                            type : T.Type) {
+  let box = T(value: ptr)
+}
+
+@inline(never)
+func createTypeWithProtocol(_ ptr: UnsafeMutablePointer<Int32>, type : InitWithPtr.Type) {
+  let box = type.self.init(value: ptr)
+}
+
+@inline(never)
+func createTypeWithProtocol(_ ptr: UnsafeMutablePointer<Int32>,
+                            type : InitWithPtr.Type,
+                            holder : InitWithBox.Type) {
```

Comment:
```suggestion
func createTypeWithProtocol(
  _ ptr: UnsafeMutablePointer<Int32>,
  type: InitWithPtr.Type,
  holder: InitWithBox.Type
) {
```

Reply:
SGTM. Is there a tool (like clang-format) for swift?